frontend/src/views/SoundEditingUpload.vue

```diff
@@ -0,0 +1,286 @@
+  user: any = null;
+  userError: any = null;
+  files: Map<File, IFileStatus> = new Map();
+  totalUploadCount: number = 0;
+  completedFileUploads: number = 0;
+
+  $refs!: {
+    myDropzone: any;
+  };
+
+  dropzoneOptions = {
+    previewTemplate: this.template(),
+    url: "localhost",
+    autoProcessQueue: false,
+    uploadMultiple: true
+  };
+
+  getFile(file: File): IFileStatus {
+    return this.files.get(file) || {};
+  }
+
+  getFiles() {
+    return Array.from(this.files);
+  }
+
+  deleteFile(file: File) {
+    this.files.delete(file);
+    this.$forceUpdate();
+  }
+
+  updateFileStatus(file: File, path: any, value: any) {
+    this.updateFileFields(file, { [path]: value });
+  }
+
+  updateFileFields(file: File, obj: IFileStatus) {
+    const selectedFile = this.getFile(file);
+    const newFile = { ...selectedFile, ...obj };
+    if (typeof obj.uploading === "boolean") {
+      this.totalUploadCount += obj.uploading ? 1 : -1;
+    }
+    this.files.set(file, newFile);
+    this.$forceUpdate();
+  }
+
+  cancelFile({ uploadTask }: IFileStatus = {}) {
+    uploadTask && uploadTask.cancel();
+  }
+
+  cancelAllFiles() {
+    this.files.forEach(status => {
+      this.cancelFile(status);
+    });
+  }
+
+  clearCompletedFiles() {
+    for (let [file, status] of this.files) {
+      if (status.complete) {
+        this.files.delete(file);
+      }
+    }
+    this.completedFileUploads = 0;
+  }
+
+  async getTask(file: File, timestamp: number) {
+    const taskId = getTaskId(file.name);
+    if (taskId) {
+      const {
+        params: { uploadCode }
+      } = this.$route;
+      this.uploadFile(
+        `sound-editing/restored/uploads/${uploadCode}/${timestamp}/${taskId}.flac`,
+        file
+      );
+    }
+  }
+
+  filesAdded(selectedFiles: File | FileList) {
+    // when a file is selected by clicking the box "selectedFiles" is an object
+    // when files are dropped selectedFiles is an array
+    setTimeout(() => this.$refs.myDropzone.removeAllFiles(), 100);
+    const timestamp = new Date().valueOf();
+    let files = Array.isArray(selectedFiles)
+      ? selectedFiles
+      : Object.values(selectedFiles);
+    files.forEach(file => {
+      this.files.set(file, {});
+      try {
+        validateFlacFile(file);
+        this.getTask(file, timestamp);
+      } catch (e) {
+        this.emitFileError(file, e.message);
+      }
+    });
+  }
+
+  emitFileError(file: File, message: string) {
+    this.updateFileStatus(file, "error", message);
+  }
+
+  getTotalUploadTime() {
+    let totalUploadTime = 0;
+    this.files.forEach(({ uploadRemainingTime: uploadTime = 0 }) => {
+      totalUploadTime += uploadTime === Infinity ? 0 : uploadTime;
+    });
+    return totalUploadTime < 60
+      ? "Less than a minute left"
+      : moment.duration(totalUploadTime, "seconds").humanize();
+  }
+
+  uploadFile(path: string, file: File) {
+    const metadata = {
+      customMetadata: {
+        fullName: file.name
+      }
+    };
```

Comment:
This metadata is not needed anymore, as the file names will not contain anything except Task ID. SO even the path could be constructed here out of the file name.